java/vector/src/main/java/org/apache/arrow/vector/compare/TypeEqualsVisitor.java

```diff
@@ -138,12 +144,12 @@ public Boolean visit(LargeListViewVector left, Void value) {
 
   private boolean compareField(Field leftField, Field rightField) {
 
-    if (leftField == rightField) {
+    if (leftField.equals(rightField)) {
```

Comment:
I _think_ this was intentional: it's a cheap "is this actually the same object" check, before doing the rest of the checks below